raft/log.go

```diff
@@ -1578,8 +1614,14 @@ func (l *Log) Heartbeat(term, commitIndex, leaderID uint64) (currentIndex uint64
 
 // RequestVote requests a vote from the log.
 func (l *Log) RequestVote(term, candidateID, lastLogIndex, lastLogTerm uint64) (peerTerm uint64, err error) {
-	l.mu.Lock()
-	defer l.mu.Unlock()
+	// Ignore if snapshotting.
```

Comment:
What will this do to leader election? It may cause leader election to be restarted, right, if not enough votes were received?